### PR TITLE
Add initial widget example

### PR DESCRIPTION
**Pull request by ncdc**
Blocked by/depends on:
- [x] https://github.com/kcp-dev/controller-runtime/pull/10
- [x] https://github.com/kcp-dev/apimachinery/pull/31
- [x] https://github.com/kcp-dev/controller-runtime/pull/12

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+generate: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
+	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
+
+.PHONY: fmt
+fmt: ## Run go fmt against code.
+	go fmt ./...
+
+.PHONY: vet
+vet: ## Run go vet against code.
+	go vet ./...
```
Comment (stevekuznetsov) [inline Makefile:63]: `go test` does this implicitly these days

File: controllers/widget_controller.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+// +kubebuilder:rbac:groups=data.my.domain,resources=widgets/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=data.my.domain,resources=widgets/finalizers,verbs=update
+
+// Reconcile TODO
+func (r *WidgetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	// Include the clusterName from req.ObjectKey in the logger, similar to the namespace and name keys that are already
+	// there.
+	logger = logger.WithValues("clusterName", req.ObjectKey.Cluster.String())
```
Comment (stevekuznetsov) [inline controllers/widget_controller.go:48]: Do we expect changes to controller-runtime making this unnecessary in user-code?

**Review by stevekuznetsov — COMMENTED** [review]
Looking good!

File: Makefile (hunk 1 above)
Comment (ncdc) [reply Makefile:63]: This was boilerplate from `kubebuilder init`

File: controllers/widget_controller.go (hunk 2 above)
Comment (ncdc) [reply controllers/widget_controller.go:48]: Maaaaaybe but not in the immediate term

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**